static/js/resize.js

```diff
@@ -49,7 +50,8 @@ function get_new_heights() {
         Number.parseInt($(".narrows_panel").css("marginTop"), 10) -
         Number.parseInt($(".narrows_panel").css("marginBottom"), 10) -
         $("#global_filters").safeOuterHeight(true) -
-        $("#streams_header").safeOuterHeight(true);
+        $("#streams_header").safeOuterHeight(true) -
+        add_streams_link_height;
```

Comment:
Can you explain the logic behind why this adjustment is correct?  I don't personally understand this `resize.js` code well, so it'd be helpful to understand that as well as how you tested this logic.

Reply:
This actually sets a max height for the `streams_ filter` list such that it does not cover the entire viewport height which is available to it after the elements above it have been rendered. This is similar to what is done in the right sidebar with the `buddy_list`. Without this, the `add streams` button went out of view on reducing the height of the viewport.